accounts/keystore/account_cache.go

```diff
@@ -260,7 +261,8 @@ func (fc *fileCache) scanFiles(keyDir string) (set.Interface, set.Interface, set
 			continue
 		}
 		filesNow.Add(path)
-		if modTime.After(prevMtime) {
+		// on macOS, we get FS notifications before the actual modTime is updated
+		if modTime.After(prevMtime) || (modTime.Equal(prevMtime) && runtime.GOOS == "darwin") {
```

Comment:
Maybe a better fix would be to use `modTime.After(prevMtime) || modTime.Equal(prevMtime)` on all platforms.
Does that sound good to you?

Reply:
Yes if we want to avoid platform specific code, I believe the perf impact is negligible